src/main/util.cpp

```diff
@@ -55,9 +55,89 @@ bool segvSpin = false;
 size_t cvc4StackSize;
 void* cvc4StackBase;
 
+/* Computes the length of a string. Safe to use in a signal handler. */
+size_t safe_strlen(const char* str) {
+  size_t l = 0;
+  while (*str) {
+    str++;
+    l++;
+  }
+  return l;
+}
+
+/* Writes a string to fd. Safe to use in a signal handler. */
+void safe_print(int fd, const char* msg) {
+  write(fd, msg, safe_strlen(msg));
```

Comment:
Not the length of the string (which is the first null terminator) but the size of the buffer as an argument. You can use this to ensure nothing overflows. ATM the safe_strlen can overflow if all 20 chars in buf get set.